### ISSUE TITLE
Adding ICES grids

### ISSUE THREAD

**Issue by James-Thorson-NOAA** [James-Thorson-NOAA/VAST#226]
Hi all,

I'm starting to add pre-built extrapolation-grids for ICES regions.  Hans Gerritsen has already provided me with shapefiles for:

```R
 [1] "./ATL-IBTS-Q1" "./ATL-IBTS-Q4" "./BITS"        "./BTS"        
 [5] "./BTS-VIIa"    "./EVHOE"       "./IE-IGFS"     "./NIGFS"      
 [9] "./NS_IBTS"     "./PT-IBTS"     "./SP-ARSA"     "./SP-NORTH"   
[13] "./SP-PORC"    
```

and I'd like to use this thread to collect feedback prior to building grids and an interface.

A few questions to start:

1.  Does IBTS Q1 and Q4 have different spatial footprints, or can I just choose one?
2.  Any objections to using a 2km by 2km grid as default?
3.  Do people every post-stratify using the "AreaName" subareas within each ICES survey?
4.  Does anyone have a total area for each ICES area, and/or subareas, to compare with the areas I get by converting from shapefiles (to check tolerance to boundary locations, projection, etc)?

Cheers,
Jim

**Comment by pdolder** [comment]
Jim,

Some comments to add to Dave's:

> Does IBTS Q1 and Q3 (sic) have different spatial footprints, or can I just choose one?

Slightly. The Q1 goes further into the Southern North Sea and into the Eastern English Channel while the Q3 does not.

> Do people every post-stratify using the "AreaName" subareas within each ICES survey?

Generally not to my understanding, these are large areas with little biological meaning. For the NS IBTS there are "roundfish areas" which are used to stratify the age-length key sampling, but specifically defined areas for each species for the indices. You can see in Figure 6.1 onward in the manual:
https://www.ices.dk/sites/pub/Publication%20Reports/ICES%20Survey%20Protocols%20(SISP)/SISP%2010%20-%20Manual%20for%20the%20International%20Bottom%20Trawl%20Surveys%20-%20Revision%20IX.pdf 

If you were to have to pick one consistent grouping, I'd say the roundfish areas would be your best bet. But even then, WGs might work things up differently, e.g. http://ices.dk/sites/pub/Publication%20Reports/Advice/2019/2019/cod.27.47d20.pdf page 5.

Paul

**Comment by James-Thorson-NOAA** [comment]
I guess my biggest question then is:

Should I proceed with an interface that allows ICES users to select one or more ICES survey areas and then VAST extrapolates density to all of that selected area, or do I need an expanded interface that allows ICES users to specify a subset of a specified ICES survey area (based on AreaName strata and/or depth, latitude, longitude) that is used for extrapolating densities when computing abundance indices?

Jim

**Comment by James-Thorson-NOAA** [comment]
See below for Dave Stoke's reply to original questions, sent by email:

Does IBTS Q1 and Q4 have different spatial footprints, or can I just choose one?
> Need to be a bit careful with the IBTS insignia. The North Sea IBTS is a broadly regular grid based effort allocation in Q1 and Q3-4. Not my neighbourhood, but scanning the maps below for North Sea allocation the English Channel at least seems to have different allocations between the 2 periods. In contrast the NE Atlantic IBTS surveys are ostensibly depth and to some extent habitat stratified with a nod to some randomness.

Any objections to using a 2km by 2km grid as default?
>None in principal, just noting that an average 30min tow at 4kts covers c.4Km in length and c.80m-125m in width.

Do people every post-stratify using the "AreaName" subareas within each ICES survey?
>Not sure what this means. NS IBTS has extended index areas for example to include additional ICES rectangles containing areas they feel have additional data, in communication with the assessment group. These index areas are catered for in the ICES database DATRAS and hard coded in to produce agreed products for assessment. They have been discussing a move to some depth/habitat stratification and experimented with post-stratification which I sbeing written up currently I think.
The stratification and combining of survey data with gear differences in the NE Atlantic area is more complex and done by folks outside of the DATRAS data products route so each species dealt with separately. Gear/selectivity difference are believed negligible in the promised land of the North Sea!

Does anyone have a total area for each ICES area, and/or subareas, to compare with the areas I get by converting from shapefiles (to check tolerance to boundary locations, projection, etc)?
>Boundaries from ICES with area in Km attached or at (http://gis.ices.dk/geonetwork/srv/eng/catalog.search#/metadata/c784a0a3-752f-4b50-b02f-f225f6c815eb)

**Comment by James-Thorson-NOAA** [comment]
As an update:  I've created a new function `convert_shapefile`: https://github.com/James-Thorson-NOAA/FishStatsUtils/blob/development/R/convert_shapefile.R

This takes as input a shapefile and makes an extrapolation-grid.  I will explore using this to save ICES shapefiles and create the extrapolation-grid at runtime with a desired resolution

**Comment by James-Thorson-NOAA** [comment]
As a final update for now:  I've added options to access and bundle output from `convert_shapefile` to create an extrapolation-grid from any of the ICES survey regions.  For example try installing the development branch of `FishStatsUtils` and running

```R
library(VAST)
Grid = make_extrapolation_info( Region="NS_IBTS" )
plot(Grid)
```

or to combine across surveys using:

```R
library(VAST)
Grid = make_extrapolation_info( Region=c("BITS","EVHOE","NS_IBTS"),
  projargs='+proj=natearth +lat_0=0 +units=km' )
plot(Grid)
```

This doesn't have any interface to extract biomass over a small subset of any given survey, hence my remaining question earlier in this thread.  Also please note that I plan to keep testing it when I find time

**Comment by pdolder** [comment]
I'll check it out once I get chance and give you some feedback here.

**Comment by James-Thorson-NOAA** [comment]
Just a quick update before closing this issue -- the shapefiles I received had the wrong '.prj' (projection) file.  I corrected that via code in the original commit, but my fix stopped working when updating R to >= 3.6.X.  I have now mannually fixed the .prj files in the development branch, it the feature appears to work for R 3.5.3, 3.6.3 and 4.0.2.  So I'm closing the issue for now.
